src/chain/CacheManager.sol

```diff
@@ -31,7 +31,7 @@ contract CacheManager {
     error BidsArePaused();
     error MakeSpaceTooLarge(uint64 size, uint64 limit);
 
-    event InsertBid(bytes32 indexed codehash, uint192 bid, uint64 size);
+    event InsertBid(bytes32 indexed codehash, address program, uint192 bid, uint64 size);
```

Comment:
```suggestion
    event InsertBid(bytes32 indexed codehash, address indexed program, uint192 bid, uint64 size);
```
maybe also index program address

Reply:
The reason why I didn't do this is because you can cache somebody else's program by using another one with the same codehash. If people search by program they might miss important events.

Reply:
make sense